BitwardenShared/UI/Auth/Login/LoginProcessor.swift

```diff
@@ -185,8 +184,29 @@ class LoginProcessor: StateProcessor<LoginState, LoginAction, LoginEffect> {
             )
             state.isLoginWithDeviceVisible = isKnownDevice
         } catch {
-            coordinator.showAlert(.networkResponseError(error))
-            services.errorReporter.log(error: error)
+            await handleErrorResponse(error)
+        }
+    }
+
+    /// Handles network error responses.
+    ///
+    /// Determines whether the Bitwarden server is official or unofficial and passes this information
+    /// along with the error to the coordinator to display an appropriate alert on the main thread.
+    /// The error is also logged using the error reporter.
+    ///
+    /// - Parameter error: The error received from the network request.
+    ///
+    private func handleErrorResponse(_ error: Error) async {
+        let isOfficialBitwardenServer = await services.configService.getConfig()?.isOfficialBitwardenServer() ?? true
+        services.errorReporter.log(error: error)
+
+        await MainActor.run {
```

Comment:
🤔 Is this needed if the processor is already `@MainActor`?

Reply:
oh yeah, 🧠 💨 on my end